# 16S

## 16S identification of your isolate

In this part of the workshop, you will use publicly available 16S sequence databases and resources to obtain an identification for your isolate.

::: { .callout-important }
Please ensure that you have downloaded the 16S sequence file for your isolate (`isolate_16S.fasta`) to a suitable location on your computer.
:::

## NCBI Blast

BLAST is a bioinformatics tool developed by [NCBI (the National Center for Biotechnology Information)](https://www.ncbi.nlm.nih.gov/) that takes as input a _query_ sequence and searches for similar sequences in a reference database. BLAST ranks matches from most (first) to least similar. The tool can be downloaded and run on your own machine, or it can be used through the [NCBI-BLAST webservice](https://blast.ncbi.nlm.nih.gov/Blast.cgi).

- [NCBI-BLAST webservice](https://blast.ncbi.nlm.nih.gov/Blast.cgi)
- [NCBI-BLAST help pages](https://blast.ncbi.nlm.nih.gov/doc/blast-help/)
- [How BLAST works](https://www.nlm.nih.gov/ncbi/workshops/2022-10_Basic-Web-BLAST/how-blast-works.html)

NCBI are the repository of record for published biological sequence data, and provide several BLAST databases, including one that contains all publicly-available 16S sequence data. You will use the [NCBI-BLAST webservice](https://blast.ncbi.nlm.nih.gov/Blast.cgi), with your isolate's 16S sequence as the query, to search this database. 

::: { .callout-tip title="BLAST reports a number of useful measures for each match"}
- Percent coverage: what proportion (percentage) of the _query_ is similar enough to the match that it has been aligned
- Percentage identity: what proportion of the aligned region is identical in both the _query_ and the reported match
- Expectation (E-value): the number of matches you would expect by chance from this database that are _at least as similar as this one_, if you used the _query_ sequence to search in a database of the same size made up of completely random sequences
:::

### Analysing your data

1. Go to the [NCBI Blast webpage](https://blast.ncbi.nlm.nih.gov/Blast.cgi). You will see the landing page.

![](assets/images/blast_16s/01-landing_page.png)

2. Select `Nucleotide BLAST`. This will take you to a new BLASTN search.

![](assets/images/blast_16s/02-blast_search_fields.png)

3. Upload your `isolate_16S.fasta` file, or copy and paste the contents of the file into the box marked `Enter accession number(s), gi(s) or FASTA sequence(s)`.

![](assets/images/blast_16s/03-blast_query.png)

4. In the section headed `Choose Search Set`, select rRNA/ITS databases. This will autopopulate the database with `16S Ribosomal RNA sequences (Bacteria and Archaea)`, NCBI's complete database of 16S rRNA sequences.

::: { .callout-caution }
NCBI do not manually curate (i.e. confirm sequence quality or correct identity) the data in their databases. Whole genome sequences are checked by classification algorithms for taxonomic placement, but 16S sequences are not. Also, NCBI contains a large amount of historical data which may have been annotated using outdated or superseded taxonomies.
:::

![](assets/images/blast_16s/04-blast_database.png)

5. Click on the `BLAST` button (towards the bottom of the page) and wait for the result to appear.

::: { .callout-important title="Questions"}
1. What organism is the top hit to your query?
2. How similar is the top hit to your query (`Per. ident`/percentage identity column)? What is the alignment coverage?
3. What are the remaining top five matches to your query?
4. How similar are the top five matches to your query sequence?
5. What conclusion would you draw about the identity of your isolate from this BLAST search?
:::

## SILVA

[SILVA](https://www.arb-silva.de/) is a curated, quality-checked database of rRNA sequence data that has been run by the [Liebniz Institute DSMZ German Collection of Microorganisms and Cell Cultures](https://www.dsmz.de/) for about two decades. The SILVA site provides a number of online tools and services, including the [ACT](https://www.arb-silva.de/aligner) service that enables users to search the database with their own rRNA sequences to find the best matches and identify their organism.

- [SILVA rRNA database project](https://www.arb-silva.de/)
- [SILVA Alignment, Classification, and Tree (ACT) webservice](https://www.arb-silva.de/aligner)
- [ACT online tutorial](https://www.arb-silva.de/documentation/act-tutorial/)

### Analysing your data

1. Go to the [SILVA ACT](https://www.arb-silva.de/aligner) (Alignment, Classification, and Tree) service

![](assets/images/silva_16s/01-silva.png)

2. Upload your `isolate_16S.fasta` file, or copy and paste the contents of the file into the box marked `Input data`.
3. Check the box marked `Search and classify`, but leave all the other options as they are.

![](assets/images/silva_16s/02-silva.png)

4. Enter a name for your job in the field marked `Job Name:` (e.g. "`BM432 Block A workshop`" or "`blood isolate`")

![](assets/images/silva_16s/03-silva.png)

5. Click on the `Run Tool` button and wait for the result to appear.

::: { .callout-note }
While the job is waiting it will appear in the `Aligner Taskmanager` as _Queued_, then _Starting_, then _Processing_. It may take a couple of minutes to produce a result.
:::

![](assets/images/silva_16s/04-silva.png)

6. When the job is complete, select your result in the `Alignment Result Table`, and click on `Display Classification` 

::: { .callout-tip }
You may need to adjust the widths of the headers in the result table to see the full classification
:::

![](assets/images/silva_16s/05-silva.png)
7. Click on `Export to CSV` (lower left of `Alignment Result Table`) to download the classification result. You can view this file in `Excel` or in a plain text editor.

::: { .callout-important title="Questions"}
1. What is the taxonomic identity of the [last common ancestor (LCA)](https://evolution.berkeley.edu/glossary/most-recent-common-ancestor/) that SILVA assigns to your isolate (`lca_tax_slv` or `LCA tax. SILVA` column)?
2. How similar (`Identity` column) does SILVA say your isolate is to that last common ancestor sequence?
:::

## Summary

You have used two different databases, and two different techniques, to assign taxonomic identity to a 16S sequence. Are the results consistent with each other?

::: { .callout-important title="Questions"}
1. Do the BLAST and SILVA analyses give the same taxonomic identity for your 16S sequence?
2. If the two results differ, how do they differ?
3. If the results differ, how important do you think is the difference between the results?
4. What is your current opinion about the identity of your isolate? Have you revised it since the BLAST search? How confident are you in the identification?
:::

::: { .callout-tip title="Continue your identification"}
Now you have a possible identification from 16S, you should continue the classification of your isolate by using Multi-Locus Sequence Typing (MLST) to try to gain a more precise taxonomic placement. Click on the link to [`MLST`](mlst.qmd) (here, or below), to get started.
:::

