# Multi-Locus Sequence Typing (MLST)

## MLST identification of your isolate

In this part of the workshop, you will use a public [Multi-Locus Sequence Typing (MLST)](https://pubmlst.org/multilocus-sequence-typing) database and resources to help identify your isolate.

::: { .callout-important }
Please ensure that you have downloaded the genome file for your isolate (`isolate_genome.fasta`) to a suitable location on your computer.
:::

::: { .callout-warning collapse="true"}
## Multi-Locus Sequence Typing (MLST)

[Multi-Locus Sequence Typing (MLST)](https://pubmlst.org/multilocus-sequence-typing) is a widely-used method for bacterial identification. It is typically more precise and has more resolving power than 16S sequence analysis, but less precise than whole-genome sequence analysis (@Maiden2013-jh).

MLST works by defining _marker sequences_ for a taxon. These are typically well-conserved ("housekeeping") genes which very relatively little between organisms in the taxon, but enough to allow discrimination between them. The number of markers varies, but is usually somewhere around seven.

Each marker sequence has many _variants_ (different sequences) within the taxon, and these are known as _alleles_. Each marker allele is given a unique number (starting at 1 and counting upwards) - its _allele number_. A single organism's _sequence type (ST)_ is determined by the list of allele numbers that it contains. Organisms with the same _sequence type_ are considered to be part of the same group.

![Schematic representation of MLST. In this typing scheme there are seven markers ("housekeeping genes"). One of these is the gene _adk_, which has a different sequence in each of the strains A, B, and C - so these have different allele numbers (1, 2, and 3). Another marker, _pdhC_ has the same sequence (allele number) in strains A and C, but a different sequence (allele number) in strain B. The strains have allele numbers: 1,1,1,1,1,1,1 (A); 2,2,2,1,2,1,2 (B); 1,3,2,2,1,3,2 (C). These lists are different, so the strains have different sequence types.](assets/images/MLST.png)
:::

## pubMLST

- Go to the [pubMLST](https://pubmlst.org/) website

![](assets/images/pubMLST/01-pubmlst.png)

- Click on `Species ID` to open the bacterial species identifier

![](assets/images/pubMLST/02-pubmlst.png)

- Click on `Identify Species` to start the identification process

![](assets/images/pubMLST/03-pubmlst.png)

- Upload your `isolate_genome.fasta` file to the server either by clicking on `Click to select or drag and drop` and using the dialogue box to find your file, or by dragging the file onto the `Select FASTA file` field.
  - Alternatively, copy and paste the file contents - the _entire genome sequence_ - into the `Enter query sequence` box.
- Click on the `Submit` button and wait for the results

::: { .callout-important title="Questions"}
1. What is the predicted taxon of your isolate's genome? What level of confidence does the pubMLST tool assign to this prediction (`Support` column)?
2. How many marker genes in your isolate had an _exact_ match in the database? Do you think this is a sufficient number for accurate taxonomic placement? Why do you think that?
3. How many different species in the database had an _exact_ match to markers in your isolate's genome?
4. _Which_ species in the database had an _exact_ match to markers in your isolate's genome? What effect does this have on your confidence in your classification?
5. What is your current opinion about the identity of your isolate? Have you revised your opinion from 16S analysis due to the MLST analysis? How confident are you in your current identification?
:::

::: { .callout-tip title="Continue your identification"}
Now you have further supporting information about the identity of your isolate, you should continue the classification of your isolate by using whole-genome comparison methods to try to pin down a more definitive taxonomic placement Click on the link to [`Whole-genome Comparison`](whole_genome.qmd) (here, or below), to get started.
:::