# Multi-Locus Sequence Typing (MLST)

## MLST identification of your isolate

In this part of the workshop, you will use a public Multi-Locus Sequence Typing (MLST) database and resources to help identify your isolate.

::: { .callout-important }
Please ensure that you have downloaded the genome file for your isolate (`isolate_genome.fasta`) to a suitable location on your computer.
:::

## pubMLST

- Go to the [pubMLST](https://pubmlst.org/) website
- Click on `Species ID` to open the bacterial species identifier
- Click on `Identify Species` to start the identification process
- Upload your `isolate_genome.fasta` file to the server either by clicking on `Click to select or drag and drop` and using the dialogue box to find your file, or by dragging the file onto the `Select FASTA file` field.
  - Alternatively, copy and paste the file contents - the _entire genome sequence_ - into the `Enter query sequence` box.
- Click on the `Submit` button and wait for the results

::: { .callout-important title="Questions"}
1. What is the predicted taxon of your isolate's genome? What level of confidence does the pubMLST tool assign to this prediction (`Support` column)?
2. Of the 53 ribosomal genes used to classify all bacteria, how many of the genes in your isolate had an _exact_ match in the database? Do you think this is a sufficient number for accurate taxonomic placement?
3. How many different species in the database had an _exact_ match to markers in your isolate's genome?
4. _Which_ species in the database had an _exact_ match to markers in your isolate's genome? What effect does this have on your confidence in your classification?
5. What is your current opinion about the identity of your isolate? Have you revised your opinion due to the MLST analysis? How confident are you in the identification?
:::

::: { .callout-tip title="Continue your identification"}
Now you have further supporting information about the identity of your isolate, you should continue the classification of your isolate by using whole-genome comparison methods to try to pin down a more definitive taxonomic placement Click on the link to [`Whole-genome Comparison`](whole-genome.qmd) (here, or below), to get started.
:::