# Summary

In this workshop, you have:

1. gained experience using 16S and whole-genome sequence data to identify, bioinformatically, an unknown organism, using:
  - two 16S databases and associated search services to identify matching sequences from other organisms
  - the main reference MLST database (pubMLST) and its search tool to obtain a classification for your organism
  - two whole-genome classification approaches to obtain taxonomic classifications for your organism
2. considered and weighed the evidence from these approaches to draw an informed conclusion about the most likely taxonomic assignment, considering:
  - the relative information used by each method
  - the quality of the database(s) used in each approach
  - the bioinformatic approach taken
3. interpreted your classification according to the rules laid down in the ICNP (International Code of Nomenclature of Prokaryotes), a.k.a. the Prokaryotic Code, and at LPSN (the List of Prokaryotic names with Standing in Nomenclature).


::: { .callout-warning title="Thank you" }
That's almost the end of the workshop. Thank you for participating. We hope you found it enjoyable and interesting, and that you now understand more about how bacterial classification works, in practice.

This year is the first presentation of this particular material, and we would be very grateful to hear feedback [by email]({{< var admin.email >}}) or through the [GitHub repository Issues page]({{< var github.issues >}}).
:::

::: { .callout-tip title="Completing the workshop" }
Please now proceed to the [BM329 MyPlace](https://classes.myplace.strath.ac.uk/course/view.php?id=21635) page and complete the formative questions, to finish the workshop.
:::