<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Leighton Pritchard">

<title>BM329 Workshop A: Microbial Identification - 5&nbsp; Whole-Genome Comparison</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./lpsn.html" rel="next">
<link href="./mlst.html" rel="prev">
<link href="./icon_32.png" rel="icon" type="image/png">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./whole_genome.html"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Whole-Genome Comparison</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">BM329 Workshop A: Microbial Identification</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/sipbs-compbiol/BM329_Block_A_workshop" rel="" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface to the 2023-24 BM329 Block A workshop</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Data</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./16s.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">16S</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./mlst.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Multi-Locus Sequence Typing (MLST)</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./whole_genome.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Whole-Genome Comparison</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lpsn.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">LPSN: the List of Prokaryotic names with Standing in Nomenclature</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./summary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Summary</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#the-type-strain-genome-server-tygs" id="toc-the-type-strain-genome-server-tygs" class="nav-link active" data-scroll-target="#the-type-strain-genome-server-tygs"><span class="header-section-number">5.1</span> The Type (Strain) Genome Server (TYGS)</a></li>
  <li><a href="#genomerxiv" id="toc-genomerxiv" class="nav-link" data-scroll-target="#genomerxiv"><span class="header-section-number">5.2</span> genomeRxiv</a></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/sipbs-compbiol/BM329_Block_A_workshop/edit/main/whole_genome.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/sipbs-compbiol/BM329_Block_A_workshop/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Whole-Genome Comparison</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>In this part of the workshop, you will use public resources for whole-genome identification and classification of prokaryotes to help identify your isolate.</p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Important
</div>
</div>
<div class="callout-body-container callout-body">
<p>Please ensure that you have downloaded the genome file for your isolate (<code>isolate_genome.fasta</code>) to a suitable location on your computer.</p>
</div>
</div>
<section id="the-type-strain-genome-server-tygs" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="the-type-strain-genome-server-tygs"><span class="header-section-number">5.1</span> The Type (Strain) Genome Server (TYGS)</h2>
<p>The <a href="https://tygs.dsmz.de/">Type (Strain) Genome Server (TYGS)</a> is provided by the <a href="https://www.dsmz.de/">Liebniz Institute DSMZ</a>, the German collection of Microbes and Cell Cultures. It provides genome-based classification of organisms by comparison against a reference database of over 20,000 sequenced bacterial <em>type strains</em>. TYGS is tightly-coupled to the authoritative <a href="https://www.bacterio.net/">LPSN (List of Prokaryotic names with Standing in Nomenclature) database</a>, to ensure nomenclaturally-correct identification</p>
<div class="callout callout-style-default callout-warning callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-2-contents" aria-controls="callout-2" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Type Strains
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-2" class="callout-2-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>A <a href="https://www.atcc.org/microbe-products/type-strains#t=productTab&amp;numberOfResults=24"><em>type strain</em></a> is the representative of a taxon: the nomenclatural “type” on which the definition of a taxon (such as species) is based. Type strains must be deposited in at least two separate public culture collections, in different countries.</p>
<p>Not all taxa (in particular <em>Candidate phyla</em>) have type strains. It is a requirement for type strains that the organism is grown and cultured, as it cannot otherwise be placed in a culture collection. But it has been estimated that not only are the majority of bacteria and archaea so far uncultured, but they may even be unculturable in the laboratory (<span class="citation" data-cites="Hofer2018">Hofer (<a href="references.html#ref-Hofer2018" role="doc-biblioref">2018</a>)</span>).</p>
</div>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-3-contents" aria-controls="callout-3" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
How TYGS Classification Works
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-3" class="callout-3-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>TYGS takes a prokaryotic genome as input, and assigns a taxonomic identity after a series of pairwise genome comparisons.</p>
<p>First, the input genome is compared to the type strain database using an approach called MASH (<span class="citation" data-cites="Ondov2016">Ondov et al. (<a href="references.html#ref-Ondov2016" role="doc-biblioref">2016</a>)</span>). The ten type strain genomes, and the ten most closely related type strains identified by 16S rDNA gene (extracted automatically from the input) similarity.</p>
<p>This set of 20 type strain genomes is then used to find the best 50 matching type strains. A phylogenetic tree is then constructed using the Genome BLAST Distance Phylogeny approach (GBDP, <span class="citation" data-cites="Henz2005-th">Henz et al. (<a href="references.html#ref-Henz2005-th" role="doc-biblioref">2005</a>)</span>), and the resulting distances used to determine the 10 closest type strain genomes for each of the user genomes. The between-genome distances for this are calculated using digital DNA-DNA hybridisation (dDDH, <span class="citation" data-cites="Meier-Kolthoff2013-ps">Meier-Kolthoff et al. (<a href="references.html#ref-Meier-Kolthoff2013-ps" role="doc-biblioref">2013</a>)</span>).</p>
<p>TYGS compiles classification output into a PDF file, and presents the resulting phylogeny on the TYGS website. All results tables and trees can be downloaded in shareable formats.</p>
</div>
</div>
</div>
<ul>
<li>Go to the <a href="https://tygs.dsmz.de/">TYGS</a> server</li>
</ul>
<p><img src="assets/images/TYGS/01-tygs.png" class="img-fluid"></p>
<ul>
<li>Click on <code>Submit your query</code> (top menu), or <code>Submit your job</code> (button) to reach the query page</li>
</ul>
<p><img src="assets/images/TYGS/02-tygs.png" class="img-fluid"></p>
<ul>
<li>Click on the <code>Browse…</code> button and navigate to your <code>isolate_genome.fasta</code> file to select it for classification.</li>
<li>Enter your email address in the <code>Provide contact details</code> field</li>
</ul>
<p><img src="assets/images/TYGS/03-tygs.png" class="img-fluid"></p>
<ul>
<li>Click on <code>Submit query</code> and wait for the results</li>
</ul>
<p><img src="assets/images/TYGS/04-tygs.png" class="img-fluid"></p>
<div class="callout callout-style-default callout-warning callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Warning
</div>
</div>
<div class="callout-body-container callout-body">
<p>This may take some time (maybe over an hour, depending on server load!), and may not be complete before the end of the workshop.</p>
<p><strong>Please move on to the next section while you are waiting.</strong></p>
</div>
</div>
<ul>
<li>When you get the result confirmation email, download and inspect the result PDF, and examine the whole-genome tree.</li>
</ul>
<p><img src="assets/images/TYGS/05-tygs.png" class="img-fluid"></p>
<div class="callout callout-style-default callout-important callout-titled" title="Questions">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Questions
</div>
</div>
<div class="callout-body-container callout-body">
<ol type="1">
<li>What is the predicted taxonomic classification of your isolate’s genome?</li>
<li>How similar is your isolate’s genome to the closest match? (the <span class="math inline">\(d_4\)</span> result is the dDDH (digital DNA-DNA hybridisation) score)</li>
<li>What are the most closely-related species and genera in the whole-genome tree? Is the distribution of genera consistent with what was previously known about <em>Ochrobactrum</em> and <em>Brucella</em> taxonomy?</li>
<li>What is your current opinion about the identity of your isolate? How confident are you in the identification?</li>
</ol>
</div>
</div>
</section>
<section id="genomerxiv" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="genomerxiv"><span class="header-section-number">5.2</span> genomeRxiv</h2>
<p><a href="http://genomerxiv.cs.vt.edu/"><code>genomeRxiv</code></a> is a recently-developed approach to bacterial classification that promises to identify and classify prokaryotic genomes quickly and accurately into categories called LINgroups (LIN: Life Identification Number). LINgroups are a taxonomy-independent, quantitative categorisation scheme that organises genome sequences by similarity in multidimensional “space”. These categorisations can then be used to relate alternative taxonomic assignments, and other annotations, to each other (<span class="citation" data-cites="Pritchard2022-ti">Pritchard et al. (<a href="references.html#ref-Pritchard2022-ti" role="doc-biblioref">2022</a>)</span>).</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-6-contents" aria-controls="callout-6" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
How <code>genomeRxiv</code> works
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-6" class="callout-6-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p><code>genomeRxiv</code> works in a similar way to <a href="https://en.wikipedia.org/wiki/Projected_coordinate_system">map grid references</a>.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="assets/images/DarronGedge-HowToReadASixFigureGridReference-Screenshot.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Example map grid references: point A has grid reference 970 (Easting) and 280 (Northing) as <em>x</em>-<em>y</em> co-ordinates, so the total reference is 970280. Similarly, point B is at 989319, and point C at 005255. All points in the map, however large or small, are uniquely assigned to a discrete location and, by subdividing the map into smaller and smaller squares (longer and longer numbers) an arbitrary level of precision can be reached</figcaption>
</figure>
</div>
<p><code>genomeRxiv</code> compares input sequences to a reference database with a very fast bioinformatics algorithm (<code>sourmash</code>) to get a set of good matches, and then refines the match with a more precise but slower algorithm (<code>ANI</code>). <code>genomeRxiv</code> then assigns a LINgroup to the genome (<span class="citation" data-cites="Tian2021-xc">Tian et al. (<a href="references.html#ref-Tian2021-xc" role="doc-biblioref">2021</a>)</span>). The LINgroup is a string of numbers, analogous to a map co-ordinate. The shorter the LINgroup, the lower the resolution of identification (Phylum, Family, etc.), and the longer the LINgroup, the finer the resolution (species, subspecies, strain, etc.).</p>
<p>The first key difference between LINgroups and map co-ordinates is that LINgroups are not co-ordinates on a two-dimensional surface, but in many-dimensional space. The second is that a single number describes the location, rather than two numbers (the “Easting” and “Northing” of map co-ordinates. Finally, grid references represent a physical space (such as the surface of the Earth), and LINgroups represent “sequence space” - which is not physical (<span class="citation" data-cites="Mazloom2022-uv">Mazloom et al. (<a href="references.html#ref-Mazloom2022-uv" role="doc-biblioref">2022</a>)</span>).</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="assets/images/genomeRxiv/projection_of_ani_distances.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Example arbitrary projection of 380 <em>Ralstonia solanacearum</em> genomes into three dimensions, based on overall genome similarity. Each sphere is a single genome. The genomes can be clustered into groups of genomes that are more similar to each other than they are to other genomes in the dataset. These clusters are assigned different colours. Each genome, such as the one indicated, has a unique co-ordinate in this “sequence space.”</figcaption>
</figure>
</div>
<p>Taxa are then defined by the volumes in space circumscribed by genomes which are examples of each taxon. When a new unknown genome is added, a LINgroup is assigned and - if it lies within a volume of space contained only by members of a single taxon (e.g.&nbsp;<em>E. coli</em>), the genome is assigned that taxon.</p>
</div>
</div>
</div>
<p>The <a href="http://genomerxiv.cs.vt.edu/"><code>genomeRxiv</code> webservice</a> allows users to input their bacterial genomes and rapidly obtain, or predict, taxonomic assignments on the basis of genome sequence.</p>
<div class="callout callout-style-default callout-warning callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Warning
</div>
</div>
<div class="callout-body-container callout-body">
<p><code>genomeRxiv</code> remains under active development and is not yet fully-released, although it is public and usable.</p>
</div>
</div>
<ul>
<li>Go to the <a href="http://genomerxiv.cs.vt.edu/">genomeRxiv</a> server</li>
</ul>
<p><img src="assets/images/genomeRxiv/01-genomeRxiv.png" class="img-fluid"> - Click on <code>Identify using a FASTA file</code></p>
<p><img src="assets/images/genomeRxiv/02-genomeRxiv.png" class="img-fluid"></p>
<ul>
<li>Either click on the <code>Sequence to be identified</code> link to bring up a dialogue box through which you can upload the <code>isolate_genome.fasta</code> file, or drag the <code>isolate_genome.fasta</code> file onto the text.</li>
<li>Click on <code>Identify</code> and wait for the results</li>
</ul>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The genomeRxiv site may ask you to allow pop-ups. You should allow the pop-ups, as these contain the identification output you need.</p>
</div>
</div>
<div class="callout callout-style-default callout-warning callout-titled" title="Interpreting genomeRxiv results">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Interpreting genomeRxiv results
</div>
</div>
<div class="callout-body-container callout-body">
<p>genomeRxiv provides three classifications:</p>
<ul>
<li><strong>Tentative LIN</strong>: this is the LINgroup specific to the submitted genome. It may or may not match an existing, previously-assigned LINgroup.</li>
<li><strong>Closest Genome</strong>: this is the LINgroup corresponding to the closest-matching genome in the genomeRxiv database. It is unlikely to match the LIN of the submitted genome exactly, even if it’s a genome from the same species.</li>
<li><strong>Member LINgroups</strong>: this is LINgroup that circumscribes a known taxon, as close to <em>and enclosing</em> the submitted genome.</li>
</ul>
</div>
</div>
<div class="callout callout-style-default callout-important callout-titled" title="Questions">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Questions
</div>
</div>
<div class="callout-body-container callout-body">
<ol type="1">
<li>What is the predicted taxonomic classification of your isolate’s genome? (Check the <code>Member LINgroups</code> result)</li>
<li>What is the taxonomic identity assigned to the most similar genome in the database?</li>
<li>How similar is your isolate’s genome to the closest match? (Look at the <code>ANI to Target</code> value)</li>
<li>What is your current opinion about the identity of your isolate? Have you modified your classification? How confident are you in the identification?</li>
</ol>
</div>
</div>
<div class="callout callout-style-default callout-tip callout-titled" title="Consider your final evaluation">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Consider your final evaluation
</div>
</div>
<div class="callout-body-container callout-body">
<p>You have now used a number of different online bioinformatics tools to obtain a possible taxonomic classification for the isolate in your blood sample. You should, by this point, have some idea of what you think the organism is, and how confident you are. Now it’s time to take a look at the official prokaryotic nomenclature database, to get a little more context around the candidate taxonomic name. Click on the link to <a href="./lpsn.html"><code>LPSN</code></a> (here, or below), to keep going.</p>
</div>
</div>


<div id="refs" class="references csl-bib-body hanging-indent" role="list" style="display: none">
<div id="ref-Henz2005-th" class="csl-entry" role="listitem">
Henz, Stefan R, Daniel H Huson, Alexander F Auch, Kay Nieselt-Struwe, and Stephan C Schuster. 2005. <span>“Whole-Genome Prokaryotic Phylogeny.”</span> <em>Bioinformatics</em> 21 (10): 2329–35.
</div>
<div id="ref-Hofer2018" class="csl-entry" role="listitem">
Hofer, Ursula. 2018. <span>“The Majority Is Uncultured.”</span> <em>Nature Reviews Microbiology</em> 16 (12): 716–17. <a href="https://doi.org/10.1038/s41579-018-0097-x">https://doi.org/10.1038/s41579-018-0097-x</a>.
</div>
<div id="ref-Mazloom2022-uv" class="csl-entry" role="listitem">
Mazloom, Reza, Leighton Pritchard, C Titus Brown, Boris A Vinatzer, and Lenwood S Heath. 2022. <span>“<span>LINgroups</span> as a Principled Approach to Compare and Integrate Multiple Bacterial Taxonomies.”</span> In <em>Proceedings of the 13th <span>ACM</span> International Conference on Bioinformatics, Computational Biology and Health Informatics</em>. New York, NY, USA: ACM.
</div>
<div id="ref-Meier-Kolthoff2013-ps" class="csl-entry" role="listitem">
Meier-Kolthoff, Jan P, Alexander F Auch, Hans-Peter Klenk, and Markus Göker. 2013. <span>“Genome Sequence-Based Species Delimitation with Confidence Intervals and Improved Distance Functions.”</span> <em>BMC Bioinformatics</em> 14 (1): 60.
</div>
<div id="ref-Ondov2016" class="csl-entry" role="listitem">
Ondov, Brian D., Todd J. Treangen, Páll Melsted, Adam B. Mallonee, Nicholas H. Bergman, Sergey Koren, and Adam M. Phillippy. 2016. <span>“Mash: Fast Genome and Metagenome Distance Estimation Using MinHash.”</span> <em>Genome Biology</em> 17 (1). <a href="https://doi.org/10.1186/s13059-016-0997-x">https://doi.org/10.1186/s13059-016-0997-x</a>.
</div>
<div id="ref-Pritchard2022-ti" class="csl-entry" role="listitem">
Pritchard, Leighton, C Titus Brown, Bailey Harrington, Lenwood S Heath, N Tessa Pierce-Ward, and Boris A Vinatzer. 2022. <span>“Could a Focus on the <span>‘Why’</span> of Taxonomy Help Taxonomy Better Respond to the Needs of Science and Society?”</span> <em>Front. Microbiol.</em> 13 (May): 887310.
</div>
<div id="ref-Tian2021-xc" class="csl-entry" role="listitem">
Tian, Long, Reza Mazloom, Lenwood S Heath, and Boris A Vinatzer. 2021. <span>“<span>LINflow</span>: A Computational Pipeline That Combines an Alignment-Free with an Alignment-Based Method to Accelerate Generation of Similarity Matrices for Prokaryotic Genomes.”</span> <em>PeerJ</em> 9 (e10906): e10906.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./mlst.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Multi-Locus Sequence Typing (MLST)</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./lpsn.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">LPSN: the List of Prokaryotic names with Standing in Nomenclature</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">These BM329 workshops were written by Leighton Pritchard and Morgan Feeney.</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">This book was built with <a href="https://quarto.org/">Quarto</a>.</div>
  </div>
</footer>



</body></html>